src/NumberTheory/QuadFormAndIsom/enumeration.jl

```diff
@@ -1451,10 +1451,62 @@ function splitting_of_hermitian_type(
   # If p divides n, then any output is still of hermitian type since taking pth
   # power decreases the order of the isometry
   if iszero(mod(n, p))
-    reps = representatives_of_hermitian_type(Lf, p, fix_root; cond=get(eiglat_cond, k, Int[-1, -1, -1]), genusDB, root_test, info_depth, discriminant_annihilator)
+    cond = unique!([get(i, k, Int[-1, -1, -1]) for i in eiglat_cond])
+    reps = representatives_of_hermitian_type(Lf, p, fix_root; cond, genusDB, root_test, info_depth, discriminant_annihilator)
     return reps
   end
 
+  reps = ZZLatWithIsom[]
+  V = _from_cyclotomic_polynomial_to_dict(characteristic_polynomial(Lf))
+  for i in keys(V)
+    V[i] = euler_phi(i)*V[i]
+  end
+  
+  # avoid overcounting
+  eiglat_cond_trimmed = Vector{Dict{Int, Vector{Int}}}()
+  for cond in eiglat_cond
+    T = Dict{Int64,Vector{Int64}}()
+    T[n] = get(cond, n, Int[-1, -1, -1])
+    T[k] = get(cond, k, Int[-1, -1, -1])
+    push!(eiglat_cond_trimmed, T)
+  end 
+  unique!(eiglat_cond_trimmed)
+
+
+  for cond in eiglat_cond_trimmed
+    condp = _conditions_after_power(cond, p)
+    if !all(V[i]==get(condp,i,[-1])[1]||-1==get(condp,i,[-1])[1] for i in keys(V))
```

Comment:
```suggestion
    if !all(get(condp, i, [-1])[1] == V[i] || get(condp, i, [-1])[1] == -1 for i in keys(V))
```
It was hard to read before, hence the suggestion